src/Microsoft.AspNetCore.SignalR.Core/HubGroupList.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public class HubGroupList : IReadOnlyCollection<ConcurrentDictionary<string, HubConnectionContext>>
+    {
+        private readonly ConcurrentDictionary<string, ConcurrentDictionary<string, HubConnectionContext>> _groups =
+            new ConcurrentDictionary<string, ConcurrentDictionary<string, HubConnectionContext>>();
+
+        public ConcurrentDictionary<string, HubConnectionContext> this[string groupName]
+        {
+            get
+            {
+                if (_groups.TryGetValue(groupName, out var group))
```

Comment:
you probably can do just
```
_groups.TryGetValue(groupName, out var group;
return group;
```
because `group` will be null if TryGetValue returns false

Reply:
Changed